ast/Trees.h

```diff
@@ -367,6 +367,27 @@ EXPRESSION(MethodDef) {
 
         // In C++20 we can replace this with bit field initialzers
         Flags() : isSelfMethod(false), isRewriterSynthesized(false), isAttrReader(false), discardDef(false) {}
+
+        Flags markSelfMethod() const {
+            Flags copy(*this);
+            copy.isSelfMethod = true;
+            return copy;
+        }
+        Flags markSynthesized() const {
+            Flags copy(*this);
+            copy.isRewriterSynthesized = true;
+            return copy;
+        }
+        Flags markAttrReader() const {
+            Flags copy(*this);
+            copy.isAttrReader = true;
+            return copy;
+        }
+        Flags markDiscardable() const {
+            Flags copy(*this);
+            copy.discardDef = true;
+            return copy;
+        }
```

Comment:
I don't really get the point of this?

Why not just

- default argument to `MK::Method` like `flags = Flags()`
- initialize flags the same way as normal:

```
Flags flags;
flags.discardDef = true
MK::Method(..., flags)
```
?

Reply:
Ah, that makes more sense.  I was thinking of builder pattern-type things, but this is probably better for C++ and Sorbet's general attitude of "struct fields are public".

Reply:
Yeah sorry I wasn't clear with what I wrote before; if you prefer the way it's done in this PR i don't oppose it strongly enough to overturn. My preference is only a slight preference.